presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java

```diff
@@ -677,6 +678,12 @@ public void testArrayPosition()
         assertFunction("ARRAY_POSITION(ARRAY [1.0, 2.0, 3.0, 4.0], 000000000000000000000003.000)", BIGINT, 3L);
         assertFunction("ARRAY_POSITION(ARRAY [1.0, 2.0, 3.0, 4.0], 3)", BIGINT, 3L);
         assertFunction("ARRAY_POSITION(ARRAY [1.0, 2.0, 3, 4.0], 4.0)", BIGINT, 4L);
+        assertFunction("ARRAY_POSITION(ARRAY [ARRAY[1]], ARRAY[1])", BIGINT, 1L);
+        // TODO: This semantic is unclear. This is not covered by the standard.
+        assertFunction("ARRAY_POSITION(ARRAY [ARRAY[null]], ARRAY[1])", BIGINT, 0L);
+        // TODO: With current semantics we cannot find a null element position, similarly to ARRAY_POSITION(ARRAY [NULL], NULL) => NULL
+        // TODO: It seems we should apply IS_DISTINCT_FROM semantics
+        assertFunction("ARRAY_POSITION(ARRAY [ARRAY[null]], ARRAY[null])", BIGINT, 0L);
```

Comment:
We need to decide on the semantics before merging this PR.

Now that we specified array equal semantics for null elements in structural types, I think the semantics here need to be specified. Previously, we delayed making a decision on these issues by simply failing these queries. But now that these queries will return a result, we have make a choice now. We want to avoid having to change the semantics in the future.

Reply:
Currently i'm following the existing semantics. We can change the semantics if we decided to do so. But that should be applied as a separate patch.

@martint